File: sbnz/sbnz-front/src/app/services/medicine/medicine.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs/Observable';
import { Medicine } from '../../model/medicine';

@Injectable({
  providedIn: 'root'
})
export class MedicineService {

  constructor(private http: HttpClient) { }

  getAllMedicine(): Observable<Medicine[]> {
    return this.http.get<Medicine[]>("http://localhost:8080/api/medicine/getAll");
  }

  createMedicine(medicine : Medicine) {
    let headers = new HttpHeaders().set('Content-Type', 'application/json; charset=utf-8');
    return this.http.post("http://localhost:8080/api/medicine/create", medicine,
    {
      headers: headers
    });
  }

  deleteMedicine(id : number) {
    return this.http.delete("http://localhost:8080/api/medicine/delete/" + id);
  }
}

File: sbnz/src/main/java/com/example/sbnz/service/MedicineService.java
package com.example.sbnz.service;

import com.example.sbnz.model.Medicine;

import java.util.Collection;

public interface MedicineService {


    Medicine create(Medicine medicine);

    Medicine delete(Long id);

    Collection<Medicine> getAll();

    Medicine findById(Long id);

    Boolean checkAllergies(Long id, Medicine medicine, String username);
}
